### PR TITLE
build: take two on upstream visibility rules

### LINKED ISSUE #12444: visibility issues after #12337
After picking up #12337, I can't get our visibility overrides going.

For context, we don't have Envoy as a git submodule and instead it we fetch it via bazel as a tarball using http_archive.

Doing something like:

```
ADDITIONAL_VISIBILITY = [
  '@foo//pinterest:__subpackages__'
]
```

in our extensions_build_config.bzl fails with:

```
ERROR: /private/var/tmp/_bazel_rgs/6fea2029bf5452e6659bcb3f721f95cc/external/envoy/BUILD:18:14: invalid package name '@foo//pinterest:__subpackages__': must start with '//'
```

cc: @lizan @alyssawilk

### PR DESCRIPTION
**Pull request by alyssawilk**
Setting things up so downstream folks can just override with visibility:public but upstream avoids core code depending on extensions.

Risk Level: medium (of breaking builds, instructions of how to unbreak included)
Testing: manually swapped to public, all works.
Docs Changes: included
Release Notes: overkill
Fixes #12444

### CODE REVIEW

**Comment by lizan** [comment]
I just merged https://github.com/envoyproxy/envoy-filter-example/pull/128, can we bump filter-example SHA in CI to see if it works?

**Comment by alyssawilk** [comment]
pass-through filter is manually tagged public (not using the default rules) so I think it's orthogonal from this change.

More interesting would be if it were private, and envoy filter example had an override to explicitly depend on it, but I agree that one should be public (and possibly moved to core but organization tbd)

**Comment by lizan** [comment]
@alyssawilk it is manually tagged public after the previous take. There were a couple PRs.

**Comment by alyssawilk** [comment]
Right, but this PR doesn't revert that being public, so I guess I'm not sure what bumping proves?
I can do it either way, I would just expect this change and that change to not interact.

File: source/extensions/extensions_build_config.bzl [hunk 1]
```diff
@@ -199,8 +199,8 @@ EXTENSIONS = {
...
-# This can be used to extend the visibility rules for Envoy extensions
-# (//:extension_config and //:extension_library in //BUILD)
-# if downstream Envoy builds need to directly reference envoy extensions.
-ADDITIONAL_VISIBILITY = [
-  ]
+
+# These can be changed to ["//visibility:public"], for  downstream builds which
+# need to directly reference Envoy extensions.
+EXTENSION_CONFIG_VISIBILITY = ["//:extension_config"]
+EXTENSION_EXTENSION_PACKAGE_VISIBILITY = ["//:extension_library"]
```
Comment (htuch) [inline source/extensions/extensions_build_config.bzl:206]: This seems reasonable. I can imagine a finer grained system, but this is fine as a starting point to limit the visibility in Envoy.

Comment (lizan) [inline source/extensions/extensions_build_config.bzl:206]: Just name this `EXTENSION_PACKAGE_VISIBILITY`?

**Comment by lizan** [comment]
@rgs1 can you test this PR with your BUILD?

**Comment by rgs1** [comment]
> @rgs1 can you test this PR with your BUILD?

Hey sorry for the late response I was OOO; yes I can give this a try. Mind merging master first tho?

**Comment by rgs1** [comment]
So with:

```
diff --git a/bazel/build_config/extensions_build_config.bzl b/bazel/build_config/extensions_build_config.bzl
index ab61ec7..5e1f725 100644
--- a/bazel/build_config/extensions_build_config.bzl
+++ b/bazel/build_config/extensions_build_config.bzl
@@ -101,3 +101,6 @@ WINDOWS_EXTENSIONS = {
 }

 ADDITIONAL_VISIBILITY = []
+
+EXTENSION_CONFIG_VISIBILITY = ["//visibility:public"]
+EXTENSION_PACKAGE_VISIBILITY = ["//visibility:public"]
```

this works for us. Thanks!

[ we do reference a lot of Envoy extensions... the gzip stuff, the crypto stuff, etc. ]

**Comment by alyssawilk** [comment]
@lizan any other thoughts?

**Comment by alyssawilk** [comment]
@lizan ping?

**Comment by alyssawilk** [comment]
cc @yanavlasov @auni53

**Pull request merged**